timeseries_storage/blueflood/blueflood.go

```diff
@@ -450,6 +450,44 @@ var samplerMap map[api.SampleMethod]sampler = map[api.SampleMethod]sampler{
 	},
 }
 
+// Blueflood will use the finest-grained resolution which doesn't exceed the slot limit.
+// Thus, if you request too many points, it will automatically downsample the data.
+func (b *Blueflood) AdjustTimerange(requested api.Timerange, slotLimit int) api.Timerange {
+	// In some cases, coarser-resolution data may have a shorter TTL.
+	// To accomodate these cases, it must be verified that the requested timerange will
+	// actually be present for the chosen resolution.
+	// TODO: figure out how to make this work with moving averages and timeshifts
```

Comment:
Yes, that is a current limitation of this design. I think the goal was that eventually the TimeseriesStorageAPI would perform upsampling as needed (for example, if asked for 30s data that's 3 days old, it would instead query the 5m data, and interpolate to get faked 30s data). This upsampling would occur at each `.fetch()` 

Adjusting timerange mostly serves to prevent the `PointFetchLimit` from being exceeded. Otherwise, you'll get an annoying "too many points to fetch" which can be fixed really easily by just using a lower resolution. Right now, this limit ignores things that change the timerange (like moving average) since it only reads the timerange before it's put into the select expression.